src/tools/query.py

```diff
@@ -96,3 +95,185 @@ def run_cluster_query(ctx: Context, query: str, **kwargs: Any) -> list[dict[str,
     except Exception as e:
         logger.error(f"Error running query: {e}")
         raise
+
+
+def get_top_longest_running_queries(
+    ctx: Context, limit: int = 10
+) -> list[dict[str, Any]]:
+    """Get the top N longest running queries from the system:completed_requests catalog.
+
+    Args:
+        limit: Number of queries to return (default: 10)
+
+    Returns:
+        List of queries with their average service time and count
+    """
+    query = """
+    SELECT statement,
+        DURATION_TO_STR(avgServiceTime) AS avgServiceTime,
+        COUNT(1) AS queries
+    FROM system:completed_requests
+    WHERE UPPER(statement) NOT LIKE 'INFER %'
+        AND UPPER(statement) NOT LIKE 'CREATE INDEX%'
+        AND UPPER(statement) NOT LIKE '% SYSTEM:%'
+    GROUP BY statement
+    LETTING avgServiceTime = AVG(STR_TO_DURATION(serviceTime))
+    ORDER BY avgServiceTime DESC
+    LIMIT $limit
+    """
+
+    return run_cluster_query(ctx, query, limit=limit)
+
+
+def get_top_most_frequent_queries(
+    ctx: Context, limit: int = 10
+) -> list[dict[str, Any]]:
+    """Get the top N most frequent queries from the system:completed_requests catalog.
+
+    Args:
+        limit: Number of queries to return (default: 10)
+
+    Returns:
+        List of queries with their frequency count
+    """
+    query = """
+    SELECT statement,
+        COUNT(1) AS queries
+    FROM system:completed_requests
+    WHERE UPPER(statement) NOT LIKE 'INFER %'
```

Comment:
For every query that we run via UI, we also run explain and advise on it.
So you might also need to filter out explain and advise for top most frequent queries